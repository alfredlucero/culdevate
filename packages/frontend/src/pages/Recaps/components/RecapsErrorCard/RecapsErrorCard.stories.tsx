import React from "react";
import { storiesOf } from "@storybook/react";
import RecapsErrorCard from "./index";

const retry = () => {
  console.log("Retry fetch!");
};

storiesOf("RecapsPage/RecapsErrorCard", module).add("Default", () => <RecapsErrorCard onRetry={retry} />);
