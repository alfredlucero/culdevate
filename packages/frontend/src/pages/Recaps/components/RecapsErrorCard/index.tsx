import React from "react";
import cn from "classnames";
import Card from "../../../../components/Card";
import Icon from "../../../../components/Icon";
import Text from "../../../../components/Text";
import Button from "../../../../components/Button";
import { CommonProps } from "../../../../components/commonProps";

interface RecapsErrorCardProps extends CommonProps {
  onRetry: () => void;
}

const RecapsErrorCard: React.FC<RecapsErrorCardProps> = ({
  onRetry,
  className = "",
  testId = "",
  ...passThroughProps
}) => {
  return (
    <Card
      className={cn("h-64", "p-6", "flex", "flex-col", "justify-center", "items-center", className)}
      testId={testId}
      {...passThroughProps}
    >
      <Icon className={cn("text-red-400", "mb-4")} variant="exclamationTriangle" size="large" />
      <Text variant="p" className={cn("text-center", "mb-6")}>
        Oops! Something went wrong with fetching your Recaps!
      </Text>
      <Button type="button" variant="primary" onClick={onRetry}>
        Retry
      </Button>
    </Card>
  );
};

export default RecapsErrorCard;
