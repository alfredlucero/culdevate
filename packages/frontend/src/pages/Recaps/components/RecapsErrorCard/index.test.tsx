import React from "react";
import { render } from "@testing-library/react";
import RecapsErrorCard from "./index";

describe("<RecapErrorCard />", () => {
  test("should render without error", () => {
    const { container } = render(
      <RecapsErrorCard onRetry={() => {}} testId="recapsErrorCardTestId" className="extra-recaps-error-card-class" />,
    );

    expect(container).toMatchSnapshot();
  });
});
