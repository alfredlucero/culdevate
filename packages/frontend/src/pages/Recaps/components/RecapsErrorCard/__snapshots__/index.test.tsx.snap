// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RecapErrorCard /> should render without error 1`] = `
<div>
  <div
    class="bg-white shadow-md rounded px-8 py-6 h-64 p-6 flex flex-col justify-center items-center extra-recaps-error-card-class"
    data-testid="recapsErrorCardTestId"
  >
    <div
      class="font-sans text-4xl text-red-400 mb-4"
      id=""
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-exclamation-triangle fa-w-18 "
        data-icon="exclamation-triangle"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 576 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M569.517 440.013C587.975 472.007 564.806 512 527.94 512H48.054c-36.937 0-59.999-40.055-41.577-71.987L246.423 23.985c18.467-32.009 64.72-31.951 83.154 0l239.94 416.028zM288 354c-25.405 0-46 20.595-46 46s20.595 46 46 46 46-20.595 46-46-20.595-46-46-46zm-43.673-165.346l7.418 136c.347 6.364 5.609 11.346 11.982 11.346h48.546c6.373 0 11.635-4.982 11.982-11.346l7.418-136c.375-6.874-5.098-12.654-11.982-12.654h-63.383c-6.884 0-12.356 5.78-11.981 12.654z"
          fill="currentColor"
        />
      </svg>
    </div>
    <p
      class="font-sans text-gray-800 text-base text-center mb-6"
    >
      Oops! Something went wrong with fetching your Recaps!
    </p>
    <button
      class="font-bold py-2 px-4 rounded font-bold py-2 px-4 rounded bg-teal-500 text-white hover:bg-teal-700 "
      type="button"
    >
      <div
        class="flex items-center"
      >
        Retry
      </div>
    </button>
  </div>
</div>
`;
